src/App.jsx

```diff
@@ -13,34 +14,89 @@ export const goodsFromServer = [
   'Jam',
   'Garlic',
 ];
+const SORT_BY_ALPHABET = 'alphabet';
+const SORT_BY_LENGTH = 'length';
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+function getPreparedGoods(goods, sortParameter) {
+  if (sortParameter) {
+    goods.sort((a, b) => {
+      switch (sortParameter) {
+        case SORT_BY_ALPHABET:
+          return a.localeCompare(b);
+        case SORT_BY_LENGTH:
+          return a.length - b.length;
+        default:
+          return 0;
+      }
+    });
+  }
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  return goods;
+}
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+export const App = () => {
+  const [isReversed, setIsReversed] = useState(false);
+  const [sortParameter, setSortParameter] = useState('');
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  const goods = getPreparedGoods([...goodsFromServer], sortParameter);
+
+  if (isReversed) {
+    goods.reverse();
+  }
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          onClick={() => {
+            setSortParameter(SORT_BY_ALPHABET);
+          }}
+          type="button"
+          className={`button is-info ${sortParameter !== SORT_BY_ALPHABET ? 'is-light' : ''}`}
```

Comment:
To set classes conditionally it's better to use [classnames](https://www.npmjs.com/package/classnames) or [clsx](https://www.npmjs.com/package/clsx) library